src/dagmc/CMakeLists.txt

```diff
@@ -13,7 +13,8 @@ set(LINK_LIBS_EXTERN_NAMES MOAB_LIBRARIES HDF5_LIBRARIES)
 include_directories(${CMAKE_BINARY_DIR}/src/dagmc)
 
 dagmc_install_library(dagmc)
-
+add_dependencies(dagmc-shared MOAB)
+target_link_libraries(dagmc-shared PUBLIC ${MOAB_INSTALL_PREFIX}/lib64/libMOAB.so)
```

Comment:
The plan is to add conditional there, to do that only if MOAB source have been downloaded and set for compilation